WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -97,6 +97,8 @@ def __init__(self, *args, **kwargs):
     self.queueCECache = {}
     self.queueSlots = {}
     self.failedQueues = defaultdict(int)
+    # failedPilotOutput stores the number of times the Site Director failed to get a given pilot output
+    self.failedPilotOutput = defaultdict(int)
```

Comment:
I know the dict gets cleared at every restart, but should there be `pop`s when the limit or success is reached? Unlike the queues, the number of pilots is potentially growing rapidly.